llvm/utils/git/github-automation.py

```diff
@@ -24,12 +24,13 @@
 
 This issue may be a good introductory issue for people new to working on LLVM. If you would like to work on this issue, your first steps are:
 
-1. In the comments of the issue, request for it to be assigned to you.
-2. Fix the issue locally.
-3. [Run the test suite](https://llvm.org/docs/TestingGuide.html#unit-and-regression-tests) locally. Remember that the subdirectories under `test/` create fine-grained testing targets, so you can e.g. use `make check-clang-ast` to only run Clang's AST tests.
-4. Create a Git commit.
-5. Run [`git clang-format HEAD~1`](https://clang.llvm.org/docs/ClangFormat.html#git-integration) to format your changes.
-6. Open a [pull request](https://github.com/llvm/llvm-project/pulls) to the [upstream repository](https://github.com/llvm/llvm-project) on GitHub. Detailed instructions can be found [in GitHub's documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request).
+1. Check that no other contributor has already been assigned to this issue. If you believe that no one is actually working on it despite an assignment, ping the person. After one week without a response, the assignee may be changed.
+1. In the comments of this issue, request for it to be assigned to you, or just create a [pull request](https://github.com/llvm/llvm-project/pulls) after following the steps below. [Mention](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this issue in the description of the pull request.
```

Comment:
Should we explicitly say that raising a PR for an issue assigned to somebody else might lead to your PR being rejected? We should perhaps get some advice from other people on this as we're creating policy here.

Reply:
The updated version doesn't sound too harsh. It's just friendly advice, no real policy so far. In the (expectedly rare) situation of a PR being created even though another person is assigned, some balancing would be required. Given the initial kind advice, it's clear that violating it is not something that will be accepted silently without explicitly writing it down.

In case you still think, this requires other opinions, who would these people be?

Reply:
@pogo59 Do you have any suggestions on this please?

Reply:
Given that the suggestion is to ping and wait a week if the issue is assigned to someone else, the risk of competing PRs is really low. And even it happens, the general "be nice" directive from the Code of Conduct would apply. I think it's fine as is.